internal/cmd/skupper/link/nonkube/link_status.go

```diff
@@ -2,28 +2,129 @@ package nonkube
+	"github.com/skupperproject/skupper/pkg/utils/validator"
 	"github.com/spf13/cobra"
 )
 
 type CmdLinkStatus struct {
-	CobraCmd  *cobra.Command
-	Flags     *common.CommandLinkStatusFlags
-	Namespace string
-	siteName  string
+	linkHandler *fs.LinkHandler
+	siteHandler *fs.SiteHandler
+	CobraCmd    *cobra.Command
+	Flags       *common.CommandLinkStatusFlags
+	namespace   string
+	linkName    string
 }
 
 func NewCmdLinkStatus() *CmdLinkStatus {
 	return &CmdLinkStatus{}
 }
 
 func (cmd *CmdLinkStatus) NewClient(cobraCommand *cobra.Command, args []string) {
-	//TODO
+	if cmd.CobraCmd != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace) != nil && cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String() != "" {
+		cmd.namespace = cmd.CobraCmd.Flag(common.FlagNameNamespace).Value.String()
+	}
+
+	cmd.linkHandler = fs.NewLinkHandler(cmd.namespace)
+	cmd.siteHandler = fs.NewSiteHandler(cmd.namespace)
+}
+
+func (cmd *CmdLinkStatus) ValidateInput(args []string) []error {
+	var validationErrors []error
+	opts := fs.GetOptions{LogWarning: false}
+	resourceStringValidator := validator.NewResourceStringValidator()
+
+	// Validate arguments name if specified
+	if len(args) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("only one argument is allowed for this command"))
+	} else if len(args) == 1 {
+		if args[0] == "" {
+			validationErrors = append(validationErrors, fmt.Errorf("link name must not be empty"))
+		} else {
+			ok, err := resourceStringValidator.Evaluate(args[0])
+			if !ok {
+				validationErrors = append(validationErrors, fmt.Errorf("link name is not valid: %s", err))
+			} else {
+				cmd.linkName = args[0]
+			}
+		}
+	}
+	// Validate that there is a link with this name in the namespace
+	if cmd.linkName != "" {
+		link, err := cmd.linkHandler.Get(cmd.linkName, opts)
+		if link == nil || err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("link %s does not exist in namespace %s", cmd.linkName, cmd.namespace))
+		}
+	}
+	return validationErrors
 }
 
-func (cmd *CmdLinkStatus) ValidateInput(args []string) []error { return nil }
-func (cmd *CmdLinkStatus) InputToOptions()                     {}
 func (cmd *CmdLinkStatus) Run() error {
-	return fmt.Errorf("command not supported by the selected platform")
+	opts := fs.GetOptions{RuntimeFirst: true, LogWarning: true}
+	if cmd.linkName == "" {
+		links, err := cmd.linkHandler.List()
+		if links == nil || err != nil {
+			fmt.Println("no links found:")
+			return err
+		}
+
+		tw := tabwriter.NewWriter(os.Stdout, 8, 8, 1, '\t', tabwriter.TabIndent)
+		_, _ = fmt.Fprintln(tw, fmt.Sprintf("%s\t%s",
+			"NAME", "STATUS"))
+		for _, link := range links {
+			status := "Not Ready"
+			if link.IsConfigured() {
+				status = "Ok"
+			}
+			name := link.Name
+			for index, endpoint := range link.Spec.Endpoints {
+				if endpoint.Name == types.InterRouterRole {
+					name = name + "-" + link.Spec.Endpoints[index].Host
```

Comment:
I belive you just need to show the name here.

Reply:
The problem with just showing the name, is that if the user wants to get the detailed output for the  skupper link status then they don't have the full name to use for that command:

skupper link status -n west -p podman
NAME				                STATUS
link-router-access-west-0.0.0.0	Not Ready

skupper link status -n west -p podman link-router-access-west-0.0.0.0
Name:		link-router-access-west
Status:		Not Ready
Cost:		2
TlsCredentials:	link-router-access-west
Endpoint:	0.0.0.0:55671

Reply:
only showing link name removed the binding-host value.